Source/WebCore/editing/Editor.cpp

```diff
@@ -2216,7 +2267,36 @@ void Editor::setComposition(const String& text, SetCompositionMode mode)
     }
 }
 
-void Editor::setComposition(const String& text, const Vector<CompositionUnderline>& underlines, const Vector<CompositionHighlight>& highlights, const HashMap<String, Vector<CharacterRange>>& annotations, unsigned selectionStart, unsigned selectionEnd)
+void Editor::setWritingSuggestion(const String& text, const CharacterRange& suggestionRange)
+{
+    Ref document = protectedDocument();
+    document->updateStyleIfNeeded();
+
+    auto inProgress = m_isHandlingAcceptedCandidate;
+
+    m_isHandlingAcceptedCandidate = true;
+
+    RefPtr selectedElement = this->selectedElement();
+    if (!selectedElement)
+        return;
+
+    auto suggestionText = text.length() >= suggestionRange.location ? text.substring(suggestionRange.location) : emptyString();
+
+    selectedElement->setWritingSuggestion(suggestionText);
+
+    // These following lines are needed for AX compatibility:
+
+    m_customCompositionAnnotations = {
+        { "NSTextCompletion"_s /* NSTextCompletionAttributeName */, { suggestionRange } },
+    };
```

Comment:
This seems very wrong to have in the platform-agnostic `Editor`. Also, I thought the AX changes were coming in a separate PR?